hub/scripts.js

```diff
@@ -9,258 +9,338 @@
+
+          // Create a specific class name for the component
+          const sectionClass = `${CONFIG.SELECTORS.NAMESPACE}--${sectionIdentifier}`;
+          // Create a placeholder element
+          // where all the transformed table markup will be added
+          const sectionMarkup = createCustomElement('div', {
+            class: sectionClass,
+          });
+
+          // Identify all the rows of the table
+          const tableRows = table.querySelectorAll('tbody tr');
+
+          tableRows.forEach((tableRow) => {
+            // For each row, create a `div` element
+            const sectionRow = createCustomElement('div', {
+              class: `${sectionClass}-row`,
+            });
+
+            // Identify all the columns of the row
+            const rowColumns = tableRow.querySelectorAll('td');
+
+            rowColumns.forEach((rowColumn) => {
+              // Get the column's content
+              const sectionEntryContent = rowColumn.innerHTML;
+
+              if (isNonEmptyString(sectionEntryContent)) {
+                // For each column that has content, create a `div` element
+                const sectionEntry = createCustomElement('div', {
+                  class: `${sectionClass}-entry`,
+                });
+
+                // Append the column content to the previously created `div` element
+                sectionEntry.innerHTML = rowColumn.innerHTML;
+                // Append the transformed column to its row
+                sectionRow.appendChild(sectionEntry);
+              }
+            });
+
+            // If the row has been populated with at least one element,
+            // coming from inner-columns, append the transformed row to the section `div`
+            if (sectionRow.hasChildNodes()) {
+              sectionMarkup.appendChild(sectionRow);
+            }
+          });
+
+          // If the section has been populated with at least one element,
+          // replace the original table with the transformed markup
+          if (sectionMarkup.hasChildNodes()) {
+            // Replace the table with the transformed markup
+            // (we can be sure the table's parent exists based on the initial selector)
+            table.parentNode.replaceChild(sectionMarkup, table);
+          }
+
+          // Measure the time, in ms, required for the table transformation
+          window.performance.measure(`tableConversionTime--${sectionIdentifier}`, startMarkerName);
+        }
+      }
+    });
+  }
+
+  function decorateEmbeds() {
+    document.querySelectorAll('a[href]').forEach(($a) => {
+      const url = new URL($a.href);
+      const usp = new URLSearchParams(url.search);
+      let embedHTML = '';
+      let type = '';
+
+      if ($a.href.startsWith('https://www.youtube.com/watch')) {
+        const vid = usp.get('v');
+
+        type = 'youtube';
+        embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
```

Comment:
future improvement: maybe we can move this to an element and add styles via a class and properties via `setAttribute`